src/app/core/public/api/impl/Circuit.ts

```diff
@@ -16,32 +17,29 @@ import {Obj}           from "../Obj";
 import {Port}          from "../Port";
 import {Wire}          from "../Wire";
 
-import {CircuitState}  from "./CircuitState";
-import {ComponentImpl} from "./Component";
-import {PortImpl}      from "./Port";
-import {WireImpl}      from "./Wire";
+import {CircuitState} from "./CircuitState";
 
 
-export abstract class CircuitImpl implements Circuit {
-    protected readonly state: CircuitState;
+export abstract class CircuitImpl implements Circuit, CircuitState {
```

Comment:
`implements Circuit` redundant here?

Reply:
It is, but idk I like the explicitness of making sure it's still a Circuit